workers/securityworker/securityworker.py

```diff
@@ -29,7 +30,15 @@ def __init__(self):
         self.add_operation(self._index_in_scanner, interval)
 
     def _index_in_scanner(self):
-        self._next_token = self._model.perform_indexing(self._next_token)
+        try:
+            with GlobalLock(self._next_token.normalize(), lock_ttl=DEFAULT_INDEXING_INTERVAL + 60):
+                self._next_token = self._model.perform_indexing(self._next_token)
+        except LockNotAcquiredException:
+            logger.debug(
+                "Could not acquire global lock for security worker indexing: %s",
+                self._next_token.normalize(),
+            )
+            self._next_token = self._next_token.next_page()
```

Comment:
So the idea is that all the workers will race on the lock, and losers race on the next token on the next interval? I think that means that means no matter how many workers there are, there will only be one started every interval.

Reply:
Right, so in theory, `x1 -> x2` would be reserved for one worker, the next one being `x2+1 -> x2+1+batch_size`.
Wondering about the interval using the batch size though. Say from range 100 to 1000, if the auto increment is still below the 1000 (i.e new manifests have ids lower than that) while a lock is on that range, then some manifests might just get missed (until is starts over from the beginning of the table again).